common/watcher/watcher.go

```diff
@@ -0,0 +1,182 @@
+package watcher
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/openmfp/golang-commons/logger"
+)
+
+// FileEventHandler handles file system events
+type FileEventHandler interface {
+	OnFileChanged(filepath string)
+	OnFileDeleted(filepath string)
+}
+
+// FileWatcher provides common file watching functionality
+type FileWatcher struct {
+	watcher *fsnotify.Watcher
+	handler FileEventHandler
+	log     *logger.Logger
+}
+
+// NewFileWatcher creates a new file watcher
+func NewFileWatcher(handler FileEventHandler, log *logger.Logger) (*FileWatcher, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create file watcher: %w", err)
+	}
+
+	return &FileWatcher{
+		watcher: watcher,
+		handler: handler,
+		log:     log,
+	}, nil
+}
+
+// WatchSingleFile watches a single file with debouncing
+func (w *FileWatcher) WatchSingleFile(ctx context.Context, filePath string, debounceMs int) error {
+	if filePath == "" {
```

Comment:
Is this a valid use case that no file is provided and it then act like a normal watcher and just waits forever?